packages/next-intl/src/extractor/extractor/MessageExtractor.test.tsx

```diff
@@ -1225,31 +448,57 @@ describe('getExtracted', () => {
     `);
   });
 
-  it('can extract with a namespace', async () => {
-    expect(
+  it('throws for dynamic description expressions', async () => {
+    await expect(async () => {
       await process(
         `
-    import {getExtracted} from 'next-intl/server';
+    import {useExtracted} from 'next-intl';
 
-    async function Component() {
-      const t = await getExtracted('ui');
-      t("Hello!");
+    function Component() {
+      const t = useExtracted();
+      const desc = 'Dynamic description';
+      t({
+        message: 'Right',
+        description: desc
+      });
     }
-          `
+  `
+      );
+    }).rejects.toThrow(
+      'Cannot extract message from dynamic expression, messages need to be statically analyzable. If you need to provide runtime values, pass them as a separate argument.'
+    );
+  });
+});
+
+describe('getExtracted', () => {
```

Comment:
@cursoragent Also move all of these tests, except for if they test anything else than input/output.